src/SelfService/Infrastructure/Persistence/AwsAccountManifestRepository.cs

```diff
@@ -0,0 +1,196 @@
+                _signature,
+                _signature,
+                new CommitOptions()
+            );
+            var result = CommandExecutor.Run("git", "push", _config.TemporaryRepoPath, 60000);
+            result.ThrowIfError();
+        }
+
+        return Task.CompletedTask;
+    }
+
+    public String ExtractManifestRef()
+    {
+        var templateManifestContent = File.ReadAllText(
+            $"{_config.TemporaryRepoPath}/00000000-0000-0000-0000-000000000000/terragrunt.hcl"
+        );
+        var match = Regex.Match(templateManifestContent, "source =.*ref=(?<ref>[^&^\"\\n\\r]*)");
+        return match.Groups["ref"].Value;
+    }
+}
+
+public class AwsAccountManifestRepositoryConfig
+{
+    public String TemporaryRepoPath { get; set; }
+    public String RemoteRepoUri { get; set; }
+    public String GitUsername { get; set; }
+    public String GitEmail { get; set; }
+    public String Branch { get; set; }
+
+    // TODO: Check and double check values
+    public AwsAccountManifestRepositoryConfig(IConfiguration configuration)
+    {
+        TemporaryRepoPath = configuration.GetValue<String>("SS_ARM_TEMPORARY_REPO_PATH") ?? "";
+        RemoteRepoUri = configuration.GetValue<String>("SS_ARM_REMOTE_REPO_URI") ?? "";
+        GitUsername = configuration.GetValue<String>("SS_ARM_GIT_USERNAME") ?? "selfservice-api";
+        GitEmail = configuration.GetValue<String>("SS_ARM_GIT_EMAIL") ?? "ssu@dfds.cloud";
+        Branch = configuration.GetValue<String>("SS_ARM_GIT_BRANCH") ?? "master";
+    }
+}
+
+public class NoOpAwsAccountManifestRepository : IAwsAccountManifestRepository
+{
+    public HashSet<string> GetAll()
+    {
+        return new HashSet<string>();
+    }
+
+    public Task Add(AwsAccountManifest manifest)
+    {
+        return Task.CompletedTask;
+    }
+}
+
+public class AwsAccountManifest
+{
+    public String Path { get; set; }
+    public AwsAccount? AwsAccount { get; set; }
+    public Capability? Capability { get; set; }
+
+    public AwsAccountManifest()
+    {
+        Path = "";
+    }
+
+    public String GenerateManifestString(string gitRef)
+    {
+        var options = "depth=1";
+        if (gitRef != "")
+        {
+            options = $"ref={gitRef}&depth=1";
+        }
+
+        // get cost centre tag
+        if (Capability?.JsonMetadata == null)
+        {
+            throw new Exception("Unable to proceed without JSON metadata for generating manifest");
+        }
+
+        // TODO: Check and double check variables!
+        return $$"""
+                terraform {
+                    source = "git::https://github.com/dfds/infrastructure-modules.git//security/org-account-context?{{options}}"
+                }
+
+                include {
+                    path = "${find_in_parent_folders()}"
+                }
+
+                inputs = {
+                    name = "dfds-gdpr-consentmanagement-maenl"
+                    email = "aws.{{Capability?.Id}}@dfds.com"
```

Comment:
Would've preferred to keep some of these in secrets, but I guess they are public now 😂